design-documents/graph-ql/coverage/shared-cart.md

```diff
@@ -1,30 +1,45 @@
+Note: This operation will only work for registered customers (valid customer token must be provided in headers).
 
- ### Scenario
+### Scenario for point 1.
  
  In the following scenario a logged in user adds a product to the cart. After logging in to his account from the smartphone the cart still contains added product (actual GraphQL queries are replaced with pseudocode for simplicity): 
  
 | Step | Device     | Operation                                                                        | Headers        | Response                                          |
 |------|------------|----------------------------------------------------------------------------------|----------------|---------------------------------------------------|
-| 1    | Laptop     | customerCartId()                                                                 | customer-token | 123e4567              |
-| 2    | Laptop     | addProductsToCart("123e4567", [{"sku": "productA", "quantity": 2}]) {cart_id}    | customer-token | 123e4567              |
-| 3    | Smartphone | customerCartId()                                                                 | customer-token | 123e4567              |
-| 4    | Smartphone | getCart("123e4567") {cart_items {sku, quantity}}                                 | customer-token | {cart_items: [{"sku": "productA", "quantity": 2]} |
+| 1    | Laptop     | customerCartId()                                                                 | customer-token | CustomerCart123e4567              |
+| 2    | Laptop     | addProductsToCart("CustomerCart123e4567", [{"sku": "productA", "quantity": 2}]) {cart_id}    | customer-token | CustomerCart123e4567              |
+| 3    | Smartphone | customerCartId()                                                                 | customer-token | CustomerCart123e4567              |
+| 4    | Smartphone | getCart("CustomerCart123e4567") {cart_items {sku, quantity}}                                 | customer-token | {cart_items: [{"sku": "productA", "quantity": 2]} |
+
 
+ 2. **Merge guest cart:** There should be a way to transfer/merge products from a guest cart to a the customer cart. To achieve this we choose to add a mutation that accepts an active guest cart Id, and reflected merged products will be shown directly in the cart response object.
+ Proposal for this scenario reflected in schema is the following query:
+```graphql
+mergeGuestIntoCustomerCart(guestCartId: String): Cart! // where Cart! is the customer cart
+```
+Note: This operation will only work for registered customers and existing guest carts (valid customer token must be provided in headers). Also, this way se segregate operations as `customerCartId()` as query and `importGuestCart()` as mutation and their intent is very clear. A developer can choose 1st or the 2nd operation depending if the user had products in the cart or not. In both situations you will get the Customer Cart . Also see alternative for #2.
```

Comment:
Please check that you use the same query/mutation names everywhere to avoid confusion